app/models/display_ad.rb

```diff
@@ -17,6 +17,7 @@ class DisplayAd < ApplicationRecord
                              inclusion: { in: ALLOWED_PLACEMENT_AREAS }
   validates :body_markdown, presence: true
   before_save :process_markdown
+  after_save :generate_display_ad_name
```

Comment:
Dropping this [conversation thread here](https://github.com/forem/forem/pull/18396#discussion_r959387296) if other folks have suggestions or thoughts:

I went with the after_save,but to be honest I'm still feeling uncertain about whether the name really needs the display_ad id. 

The other option would be to just generate a random name before saving the Display Ad if the admin does not provide a name. That way, we won't have to make that second update update after we save which may be more optimal. However, I suspect that in most cases the admin will provide a name and so I'm trying not to overthink this detail but if other folks reviewing the PR feel more strongly about which way is the better way, I'm open to changing it accordingly.